packages/next/src/client/components/params.ts

```diff
@@ -1 +1,88 @@
+import { trackDynamicDataAccessed } from '../../server/app-render/dynamic-rendering'
+import { ReflectAdapter } from '../../server/web/spec-extension/adapters/reflect'
+import { getRouteMatcher } from '../../shared/lib/router/utils/route-matcher'
+import { getRouteRegex } from '../../shared/lib/router/utils/route-regex'
+import { staticGenerationAsyncStorage } from './static-generation-async-storage.external'
+
+export type UnknownRouteParams =
+  | ReadonlySet<string>
+  | ReadonlyMap<string, string | string[]>
+  | undefined
```

Comment:
since this is presumably carried on a store using `null` instead of `undefined` to represent no params is preferable because null and objects have the same type and v8 can optimize this better